### PR TITLE
feat: expose rate limits

### PR DESCRIPTION
**Pull request by peterdeme**
- Add rate limit to response object

### CODE REVIEW

**Comment by ffenix113** [comment]
I just opened laptop to do just this! 😅

Let's also add headers. What do you think about having some class for rate limiter info, just like in Python SDK?

**Review by ferhatelmas — APPROVED** [review]

**Comment by peterdeme** [comment]
@ffenix113 done

File: stream-chat.gemspec [hunk 1]
```diff
@@ -12,9 +12,9 @@ Gem::Specification.new do |gem|
   gem.summary = 'The low level client for serverside calls for Stream Chat.'
   gem.email = 'support@getstream.io'
   gem.homepage = 'http://github.com/GetStream/stream-chat-ruby'
-  gem.authors = ['Mircea Cosbuc']
+  gem.authors = ['Stream.io']
```
Comment (ferhatelmas) [inline stream-chat.gemspec:15]: maybe `getstream.io`

**Review by ferhatelmas — APPROVED** [review]

File: stream-chat.gemspec (hunk 1 above)
Comment (peterdeme) [reply stream-chat.gemspec]: ✅

File: lib/stream-chat/stream_response.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  class StreamResponse < Hash
+    attr_reader :rate_limit
+    attr_reader :status_code
+    attr_reader :headers
+
+    def initialize(hash, response)
+      super(nil)
+      merge!(hash)
+
+      if response.headers.key?('X-Ratelimit-Limit') # rubocop:disable Style/GuardClause
```
Comment (ffenix113) [inline lib/stream-chat/stream_response.rb:16]: Rubocop does not point this as an error when I run `bundle exec rubocop` locally. Am I missing something?

I just want to understand why we need the comment here.

Comment (ferhatelmas) [reply lib/stream-chat/stream_response.rb:16]: do you have latest rubocop locally? CI installs fresh everytime

Comment (peterdeme) [reply lib/stream-chat/stream_response.rb]: this lint ignore was for a previous version of this code (where rate_limit was a dictionary, not a class). now it works fine without it, so I just removed it.

Comment (ffenix113) [reply lib/stream-chat/stream_response.rb]: Superb, thanks!

**Review by ffenix113 — APPROVED** [review]
Good job!

File: lib/stream-chat/stream_response.rb (hunk 2 above)
Comment (ferhatelmas) [reply lib/stream-chat/stream_response.rb]: Awesome 😄

**Pull request merged**